src/Bridge/Symfony/Bundle/Test/ApiTestAssertionsTrait.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Symfony\Bundle\Test;
+
+use ApiPlatform\Core\Bridge\Symfony\Bundle\Test\Constraint\ArraySubset;
+use ApiPlatform\Core\Bridge\Symfony\Bundle\Test\Constraint\MatchesJsonSchema;
+use PHPUnit\Framework\ExpectationFailedException;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/**
+ * @see \Symfony\Bundle\FrameworkBundle\Test\WebTestAssertionsTrait
+ *
+ * @experimental
+ */
+trait ApiTestAssertionsTrait
+{
+    use BrowserKitAssertionsTrait;
+
+    /**
+     * Asserts that the retrieved JSON contains has the specified subset.
+     * This method delegates to self::assertArraySubset().
+     *
+     * @throws \Symfony\Contracts\HttpClient\Exception\ClientExceptionInterface
+     * @throws \Symfony\Contracts\HttpClient\Exception\DecodingExceptionInterface
+     * @throws \Symfony\Contracts\HttpClient\Exception\RedirectionExceptionInterface
+     * @throws \Symfony\Contracts\HttpClient\Exception\ServerExceptionInterface
+     * @throws \Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface
+     */
+    public static function assertJsonContains(array $subset, bool $checkForObjectIdentity = true, string $message = ''): void
+    {
+        static::assertArraySubset($subset, self::getHttpResponse()->toArray(false), $checkForObjectIdentity, $message);
+    }
+
+    /**
+     * Asserts that the retrieved JSON is equal to the following array.
+     * Both values are canonicalized before the comparision.
+     */
+    public static function assertJsonEquals(array $json, string $message = ''): void
+    {
+        static::assertEqualsCanonicalizing($json, self::getHttpResponse()->toArray(false), $message);
```

Comment:
Because usually it's more convenient to write a PHP array than a JSON string. What would be the benefit of using `assertJsonStringEqualsJsonString`?

Reply:
I don't know how work `assertEqualsCanonicalizing` but the benefit I see is that when a test fail using `assertJsonStringEqualsJsonString`, the output error shows exactly which line of the JSON is different between real response and expected.

Reply:
AFAIK it displays a diff between both decoded PHP arrays, but between the original JSON strings. If I'm right, `assertJsonStringEqualsJsonString` does the same.

Reply:
> Because usually it's more convenient to write a PHP array than a JSON string.

@dunglas That's not always the case. A JSON string in a heredoc is more readable in my opinion.

Reply:
But sure, one could always wrap it in `json_decode(..., true)`

Reply:
Maybe could we use both, depending if the input is an array or a string?

Reply:
https://github.com/api-platform/core/pull/2940